fuse/data/tokenizers/modular_tokenizer/inject_utils.py

```diff
@@ -209,24 +197,52 @@ def prepare_info_for_model_step(
             elif tokenizer_name.startswith("VECTORS_"):
                 raise NotImplementedError
             else:
-                prev_index_end += len(curr_placeholder_encoding.ids)
-
-        if len(scalars_indices) > 0:
-            scalars_indices = torch.concat(scalars_indices)
-            scalars_values = torch.concat(scalars_values)
-        else:
-            scalars_indices = None
-            scalars_values = None
-
-        if len(scalars_masked_indices) > 0:
-            scalars_masked_indices = torch.tensor(
-                scalars_masked_indices, dtype=torch.int64
+                # prev_index_end += len(curr_placeholder_encoding.ids)
+                curr_scalar_values = torch.full(
+                    (len(curr_placeholder_encoding.ids),),
+                    fill_value=scalar_default_unfound_value,
+                )
+                all_scalars_values.append(curr_scalar_values)
+                all_scalars_valid_mask.append(
+                    torch.full_like(
+                        curr_scalar_values, fill_value=False, dtype=torch.bool
+                    )
+                )
+
+        all_scalars_values = torch.concat(all_scalars_values)
+        all_scalars_valid_mask = torch.concat(all_scalars_valid_mask)
+
+        assert all_scalars_values.shape == all_scalars_valid_mask.shape
+
+        # pad if needed
+        full_query_len = len(token_ids)
+        if full_query_len > all_scalars_values.shape[0]:
+            pad_len = full_query_len - all_scalars_values.shape[0]
+            all_scalars_values = torch.concat(
+                [
+                    all_scalars_values,
+                    torch.full(
+                        (pad_len,),
+                        fill_value=scalar_default_unfound_value,
+                        dtype=all_scalars_values.dtype,
+                    ),
+                ]
+            )
+            all_scalars_valid_mask = torch.concat(
+                [
+                    all_scalars_valid_mask,
+                    torch.full(
+                        (pad_len,), fill_value=False, dtype=all_scalars_valid_mask.dtype
+                    ),
+                ]
             )
-        else:
-            scalars_masked_indices = None
+        elif full_query_len > all_scalars_values.shape[0]:
```

Comment:
@mosheraboh see here, related to what we talked about.

I'll try to add more unit tests with interesting cases by the end of this week